core/src/torii/routing.rs

```diff
@@ -167,10 +167,21 @@ async fn handle_subscription(events: EventsSender, stream: WebSocket) -> eyre::R
     let mut consumer = Consumer::new(stream).await?;
 
     loop {
-        let event = events.recv().await?;
-
-        iroha_logger::trace!(?event);
-        consumer.consume(event).await?;
+        tokio::select! {
+            message_opt = consumer.stream_mut().next() => {
+                if let Some(message) = message_opt {
+                    if message?.is_close() {
+                        return Ok(());
+                    }
+                }
+            }
+            event = events.recv() => {
+                let event = event?;
+                iroha_logger::trace!(?event);
+                consumer.consume(event).await?;
```

Comment:
since the branches are run concurrently, it's still possible that the code will return closed stream error here because it polls the underlying stream

Reply:
if this is correct, the same is true for `Consumer::new`